### PR TITLE
refactor: FloatButton support css var

### PR DESCRIPTION
**Pull request by JarvisArt**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [x] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | FloatButton support css variables. |
| 🇨🇳 Chinese | FloatButton 组件支持 css 变量。 |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 808dbd3</samp>

This pull request refactors the float button component and its group variant to use CSS variables for styling and customization. It introduces a new hook `useCSSVar` and a new function `wrapCSSVar` to register and apply the CSS variables. It also enhances the style and customization of the component by using the `@ant-design/cssinjs` library and the `zIndexPopup` token value. It adds a new file `components/float-button/style/cssVar.ts` to export the `useCSSVar` hook.

### <samp>🤖 Generated by Copilot at 808dbd3</samp>

*  Replace `wrapSSR` function with `wrapCSSVar` function and destructure `hashId` from `useStyle` hook in `FloatButton` and `FloatButtonGroup` components to use CSS variables ([link](https://github.com/ant-design/ant-design/pull/45861/files?diff=unified&w=0#diff-28edfe59d92b75ce5389381bb8b05943da4ec4f6adc98fda85791009ff922d4bL40-R42),[link](https://github.com/ant-design/ant-design/pull/45861/files?diff=unified&w=0#diff-28edfe59d92b75ce5389381bb8b05943da4ec4f6adc98fda85791009ff922d4bL95-R97),[link](https://github.com/ant-design/ant-design/pull/45861/files?diff=unified&w=0#diff-e5d88d36fdd37df3e6d7bd4b14ab20798187137dd85d28acaee0a597cd9a09aeL35-R37),[link](https://github.com/ant-design/ant-design/pull/45861/files?diff=unified&w=0#diff-e5d88d36fdd37df3e6d7bd4b14ab20798187137dd85d28acaee0a597cd9a09aeL107-R109))
* Import `useCSSVar` hook from `./style/cssVar` in `FloatButton` and `FloatButtonGroup` components to register and use CSS variables ([link](https://github.com/ant-design/ant-design/pull/45861/files?diff=unified&w=0#diff-28edfe59d92b75ce5389381bb8b05943da4ec4f6adc98fda85791009ff922d4bR21),[link](https://github.com/ant-design/ant-design/pull/45861/files?diff=unified&w=0#diff-e5d88d36fdd37df3e6d7bd4b14ab20798187137dd85d28acaee0a597cd9a09aeR15))
* Add `components/float-button/style/cssVar.ts` file to export `useCSSVar` hook that uses `genCSSVarRegister` and `prepareComponentToken` functions to create a CSS variable provider and consumer for the float button component ([link](https://github.com/ant-design/ant-design/pull/45861/files?diff=unified&w=0#diff-ed436ba170837b14575a3b22024a0dc456409edfb9e4d7e4f62a443c3399abdeR1-R4))
* Use `unit` function from `@ant-design/cssinjs` to convert numeric token values to CSS unit strings in `initFloatButtonGroupMotion` and `sharedFloatButtonStyle` functions ([link](https://github.com/ant-design/ant-design/pull/45861/files?diff=unified&w=0#diff-fc14ffe666c4ac5e79188e926f924116e9e8a9c57e2717b2a490d878b09e72d9L38-R50),[link](https://github.com/ant-design/ant-design/pull/45861/files?diff=unified&w=0#diff-fc14ffe666c4ac5e79188e926f924116e9e8a9c57e2717b2a490d878b09e72d9L56-R68),[link](https://github.com/ant-design/ant-design/pull/45861/files?diff=unified&w=0#diff-fc14ffe666c4ac5e79188e926f924116e9e8a9c57e2717b2a490d878b09e72d9L149-R167),[link](https://github.com/ant-design/ant-design/pull/45861/files?diff=unified&w=0#diff-fc14ffe666c4ac5e79188e926f924116e9e8a9c57e2717b2a490d878b09e72d9L176-R189),[link](https://github.com/ant-design/ant-design/pull/45861/files?diff=unified&w=0#diff-fc14ffe666c4ac5e79188e926f924116e9e8a9c57e2717b2a490d878b09e72d9L265-R281),[link](https://github.com/ant-design/ant-design/pull/45861/files?diff=unified&w=0#diff-fc14ffe666c4ac5e79188e926f924116e9e8a9c57e2717b2a490d878b09e72d9L324-R340),[link](https://github.com/ant-design/ant-design/pull/45861/files?diff=unified&w=0#diff-fc14ffe666c4ac5e79188e926f924116e9e8a9c57e2717b2a490d878b09e72d9L346-R362))
* Use `calc` function from `@ant-design/cssinjs` to create CSS `calc` expressions for token values that accept custom CSS units or expressions in `floatButtonGroupStyle`, `sharedFloatButtonStyle`, and default export functions ([link](https://github.com/ant-design/ant-design/pull/45861/files?diff=unified&w=0#diff-fc14ffe666c4ac5e79188e926f924116e9e8a9c57e2717b2a490d878b09e72d9R108),[link](https://github.com/ant-design/ant-design/pull/45861/files?diff=unified&w=0#diff-fc14ffe666c4ac5e79188e926f924116e9e8a9c57e2717b2a490d878b09e72d9R228),[link](https://github.com/ant-design/ant-design/pull/45861/files?diff=unified&w=0#diff-fc14ffe666c4ac5e79188e926f924116e9e8a9c57e2717b2a490d878b09e72d9L246-R261),[link](https://github.com/ant-design/ant-design/pull/45861/files?diff=unified&w=0#diff-fc14ffe666c4ac5e79188e926f924116e9e8a9c57e2717b2a490d878b09e72d9L357-R418))
* Use `zIndexPopup` token value instead of hard-coded `99` value for `zIndex` property of `FloatButton` and `FloatButtonGroup` components ([link](https://github.com/ant-design/ant-design/pull/45861/files?diff=unified&w=0#diff-fc14ffe666c4ac5e79188e926f924116e9e8a9c57e2717b2a490d878b09e72d9L101-R114),[link](https://github.com/ant-design/ant-design/pull/45861/files?diff=unified&w=0#diff-fc14ffe666c4ac5e79188e926f924116e9e8a9c57e2717b2a490d878b09e72d9L222-R236))
* Add comments to `ComponentToken` interface to describe the meaning and usage of each token value ([link](https://github.com/ant-design/ant-design/pull/45861/files?diff=unified&w=0#diff-fc14ffe666c4ac5e79188e926f924116e9e8a9c57e2717b2a490d878b09e72d9L12-R26))
* Modify `FloatButtonToken` interface to make some token values accept either a number or a string, such as `floatButtonIconSize`, `floatButtonBodySize`, and `badgeOffset` ([link](https://github.com/ant-design/ant-design/pull/45861/files?diff=unified&w=0#diff-fc14ffe666c4ac5e79188e926f924116e9e8a9c57e2717b2a490d878b09e72d9L21-R38))

### CODE REVIEW

File: components/float-button/FloatButtonGroup.tsx [hunk 1]
```diff
@@ -32,7 +33,8 @@ const FloatButtonGroup: React.FC<FloatButtonGroupProps> = (props) => {
 
   const { direction, getPrefixCls } = useContext<ConfigConsumerProps>(ConfigContext);
   const prefixCls = getPrefixCls(floatButtonPrefixCls, customizePrefixCls);
-  const [wrapSSR, hashId] = useStyle(prefixCls);
+  const [, hashId] = useStyle(prefixCls);
+  const wrapCSSVar = useCSSVar(prefixCls);
```
Comment (MadCcc) [inline components/float-button/FloatButtonGroup.tsx:37]: 用 useCSSVarCls 取一个 roolCls 吧，FloatButton 那里也是。
这里 prefixCls 不是最外层，不会作用到最外层元素

File: components/float-button/style/index.ts [hunk 2]
```diff
@@ -1,15 +1,29 @@
...
+  /**
+   * @desc 圆形按钮中徽章点的偏移量
+   * @descEN Offset of the badge dot in a circular button
+   */
+  dotOffsetInCircle: number;
+  /**
+   * @desc 方形按钮中徽章点的偏移量
+   * @descEN Offset of the badge dot in a square button
+   */
+  dotOffsetInSquare: number;
```
Comment (MadCcc) [inline components/float-button/style/index.ts:26]: 这两个打上 @internal 或者直接移到 FloatButtonToken 里吧，不对外透出

File: components/float-button/FloatButtonGroup.tsx (hunk 1 above)
Comment (JarvisArt) [reply components/float-button/FloatButtonGroup.tsx]: done~

File: components/float-button/style/index.ts [hunk 3]
```diff
@@ -212,14 +225,15 @@ const sharedFloatButtonStyle: GenerateStyle<FloatButtonToken, CSSObject> = (toke
...
+    calc,
   } = token;
   return {
     [componentCls]: {
       ...resetComponent(token),
       border: 'none',
       position: 'fixed',
       cursor: 'pointer',
-      zIndex: 99,
+      zIndex: token.zIndexPopup,
```
Comment (MadCcc) [inline components/float-button/style/index.ts:236]: 这个建议不改，容易 break，其他的也是

**Comment by MadCcc** [comment]
https://codesandbox.io/s/antd-reproduction-template-forked-kn3jll?file=/index.js

File: components/float-button/style/index.ts (hunk 3 above)
Comment (JarvisArt) [reply components/float-button/style/index.ts:236]: 要不把 zIndexPopup 删了？因为现在这个token也没用到。

**Review by MadCcc — APPROVED** [review]

**Pull request merged**